components/brave_sync/brave_profile_sync_service_impl.h

```diff
@@ -215,6 +221,13 @@ class BraveProfileSyncServiceImpl
 
   bool IsSQSReady() const;
 
+  // Methods to run specified actions right after bookmark model was loaded
+  void OnSetupSyncNewToSyncBkmLoaded();
+  void OnSetupSyncHaveCodeBkmLoaded(const Uint8Array& seed);
+  void OnSyncReadyBkmLoaded();
+  void SendSyncRecordsBkmLoaded(const std::string& category_name,
```

Comment:
we shouldn't queue `SendSyncRecords` either, if bookmark model is not loaded, commit message won't be constructed and `OnNudgeSyncCycle` won't be triggered